kafka-connect-fitbit-source/src/main/java/org/radarbase/connect/rest/fitbit/user/ServiceUserRepository.java

```diff
@@ -123,26 +123,49 @@ public void initialize(RestSourceConnectorConfig config) {
         logger.error("Failed to initially get users from repository", ex);
       }
     }
-    return this.timedCachedUsers.stream();
+    return this.timedCachedUsers.stream()
+        .filter(User::isComplete);
   }
 
   @Override
   public String getAccessToken(User user) throws IOException, NotAuthorizedException {
+    if (!user.isAuthorized()) {
```

Comment:
Just wondering, How would this prevent temporary failures when refreshing tokens being completely removed out of the system? In this case the researchers have to recruit again.

Reply:
Indeed, but temporary failures are handled by the rest source authoriser. This update is just to ensure that unauthorised users do not pulled again and again. If the rest source authoriser fails to update a token because it gets a 400, 401 or 403 error from the backend, it will permanently mark the user as unauthorised, until the user is manually authorised again.